src/components/hooks/useSearch.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react'
+import useExercise from './useExercise'
+
+export default function useSearch() {
+  const [search, setSearch] = useState('')
+  const { exercises } = useExercise()
+
+  const searchedExercise = exercises.filter(exercise =>
+    exercise.name
+      .toLowerCase()
+      .trim()
+      .includes(search)
+  )
```

Comment:
Nice chaining! But maybe this should be wrapped in a useEffect hook with dependecies on exercises and search